### PR TITLE
[FIXED JENKINS-38721, JENKINS-37282] - NPE in "CauseOfInterruption.UserInterruption" when user is missing

### PR DESCRIPTION
**Pull request by oleg-nenashev**
It is a regression introduced by me in #2450 for [JENKINS-36594](https://issues.jenkins-ci.org/browse/JENKINS-36594). Impacts Pipeline, hence @reviewbybees

I decided to make the original method Nonnull again since there is a risk of API misusages elsewhere.

https://issues.jenkins-ci.org/browse/JENKINS-38721
https://issues.jenkins-ci.org/browse/JENKINS-37282

### CODE REVIEW

File: core/src/main/java/jenkins/model/CauseOfInterruption.java [hunk 1]
```diff
@@ -74,18 +75,46 @@ public void print(TaskListener listener) {
...
+        public String getUserId() {
+            return user;
+        }
+        
+        /**
+         * Gets user, who caused the interruption.
+         * @return User instance if it can be located.
+         *         Result of {@link User#getUnknown()} otherwise
+         */
+        @Nonnull
```
Comment (oleg-nenashev) [inline core/src/main/java/jenkins/model/CauseOfInterruption.java]: Maybe it worth deprecating it

File: core/src/main/resources/jenkins/model/CauseOfInterruption/UserInterruption/summary_zh_TW.properties [hunk 2]
```diff
@@ -22,3 +22,4 @@
 # THE SOFTWARE.
 
 blurb=\u7531\u4f7f\u7528\u8005 <a href="{1}">{0}</a> \u4e2d\u6b62
+userNotFound=\u7531\u4f7f\u7528\u8005 {0} \u4e2d\u6b62
```
Comment (jglick) [inline core/src/main/resources/jenkins/model/CauseOfInterruption/UserInterruption/summary_zh_TW.properties]: Where are you getting this from? Best not to add language keys unless you sure fluent.

**Review by jglick — APPROVED** [review]

File: core/src/main/resources/jenkins/model/CauseOfInterruption/UserInterruption/summary_zh_TW.properties (hunk 2 above)
Comment (oleg-nenashev) [inline core/src/main/resources/jenkins/model/CauseOfInterruption/UserInterruption/summary_zh_TW.properties]: It is same as a text above, but just without hyperlink. so I think it's fine

**Comment by oleg-nenashev** [comment]
@jglick is a CloudBees employee, hence @reviewbybees done

**Review by batmat — APPROVED** [review]
:+1:  the translations are impressive Oleg ;-). At least, even if not perfect, they're unlikely to be swear-words like we had in the past ;-) and :bee: even if I'm out of work time

File: core/src/main/java/jenkins/model/CauseOfInterruption.java [hunk 3]
```diff
@@ -74,18 +75,46 @@ public void print(TaskListener listener) {
...
+        }
+        
+        /**
+         * Gets user, who caused the interruption.
+         * @return User or {@code null} if it has not been found
+         * @since TODO
+         */
+        @CheckForNull
+        public User getUserOrNull() {
             return User.get(user, false, Collections.emptyMap());
```
Comment (jglick) [inline core/src/main/java/jenkins/model/CauseOfInterruption.java]: I think this should be

```java
return User.getById(user, false);
```

**Comment by Xyaren** [comment]
Will this be backported for LTS ?

**Comment by oleg-nenashev** [comment]
@Xyaren Yes, the fix has been already backported to 2.19.3. The ETA is this week unless I've missed something

**Comment by Xyaren** [comment]
We're already on 2.19.3. 
The fix is not appearing in https://jenkins.io/changelog-stable/.
That is why I asked.

**Comment by jglick** [comment]
It is in 2.19.4.

**Pull request merged**